### DISCUSSION TITLE
Cold start time 7-10 seconds

### DISCUSSION THREAD

**Discussion by ido-postelnik** [vercel/community#307] [Help] [answered]
Hey,
Can it be that the Cold start will take around 7-10 seconds?
I assume it is a cold start issue because it happens only at the first time, and from then it takes less than a second to load the Next pages.

Additional information:
When It happens, it's like the browser does not respond for the first seconds at all.
There is even no network traffic.. like it's not loading.
After 7-10 seconds end, it starts to load, and only then I see my spinner being triggered from the "routeChangeStart".

I'm using "getServerSideProps", and nothing special in terms of 3rd party packages.

Any help would be appreciated!

**Comment by leerob** [comment, marked as answer]
A [cold start](https://vercel.com/docs/concepts/functions/conceptual-model#cold-and-hot-boots) is not 7-10 seconds. It's usually around ~250-1000ms on Lambda, but it's hard to measure an exact number. The underlying cause is actually likely from the contents of `getServerSideProps` and what you're doing for server-side rendering. You have a few options:

1. Move the data you're fetching closer to your function (e.g. putting them in the same region)
2. Adding caching for your data to be closer to your function
3. Reducing the size of the data retrieved inside `getServerSideProps`

**Related:** you might be able to mitigate this consistently if using [Incremental Static Regeneration](https://vercel.com/docs/concepts/next.js/incremental-static-regeneration).

**Reply by kachkaev** [reply]
Our app‘s cold start for `getServerSideProps` is ~10 seconds at the moment. We are using Next 12.1. See https://github.com/vercel/community/discussions/427#discussioncomment-2332404 for details and the workaround.

**Reply by marcusglowe** [reply]
Running into the same issue - but this is with a `getServerSideProps` that only calls another service (which returns in ~1-200ms). No db connections, same region, etc. Timing the function execution reveals it's purely due to cold start.

Are there any debugging tips?

**Reply by kachkaev** [reply]
@marcusglowe see https://github.com/vercel/community/discussions/496. Your case might be different, but at least there's some extra data that might be relevant. Good luck! Curious to learn the outcomes of your investigation!

**Reply by terryli0095** [reply]
It's 5-10 seconds even when getServerSideProps is empty many seem to face the same issue
https://github.com/vercel/next.js/discussions/12447 @leerob

**Reply by leerob** [reply]
What is your tech stack? Can you share a CodeSandbox?

**Reply by moegts** [reply]
I think this has to deal with the amount of space you provide for each team in your server or something
because this was not the case for my old projects and once I start adding more and bigger size projects it start happening on all sites
with that being said, I am moving all my projects to AWS since I have been trying to know anything and tried all your solitons on the internet or the on emails nothing change anything and it is annoying

**Reply by denvudd** [reply]
I am experiencing a similar problem when deploying to Vercel. My "health check" example is quite similar - it's just a function in `pages/api/hello` that outputs JSON. The loading time is around 2 seconds if it's a cold start and less than 500 milliseconds if it's a warm start. The situation is the same with getServerSideProps (even worse). I've been trying to find the problem for a long time and can't find a solution. In my working project, getServerSideProps takes 7 seconds, which is not normal. I have found that many people are experiencing a similar issue but with different conditions.
